spec/index.bs

```diff
@@ -305,6 +305,9 @@ dictionary GPUBindGroupLayoutBinding {
     required u32 binding;
     required GPUShaderStageFlags visibility;
     required GPUBindingType type;
+    // For texture bindings only, we need to know the dimensions at the
+    // bind group creation time and validate the created bind groups accordingly.
+    GPUTextureViewDimension? textureDimension;
```

Comment:
Is the `?` needed? I think it can be removed.

Reply:
it's not needed for buffer type bindings

Reply:
Right, but since it's not `required`, it should be optional by default. I don't think the extra `?` is needed because that just makes it optional AND nullable.

Reply:
I think we have problems with this elsewhere in the file too, actually.

Reply:
Hm, that was not quite right. The WebIDL spec explicitly allows nullable enumerations inside dictionaries. I'm not sure exactly what the difference is between:

```webidl
GPUTextureViewDimension? textureDimension = null;
```

and

```webidl
GPUTextureViewDimension textureDimension;
```

except that the first will convert undefined to null, and the second will not. The second will also not allow null, I think.

That said, I think one of those two should be used, rather than what we have now.

Reply:
Wouldn't `GPUTextureViewDimension?` default to null?

Reply:
From what I understand (as in very little) in WebIDL `GPUTetureViewDimension? textureDimension` would default to being undefined. Basically this is a Rust `Option<Option<GPUTextureViewDimension>>` and not `Option<GPUTextureDimension>`.

Either of Kai's suggestions would work to be an `Option<GPUTextureViewDimension>` so I hae a slight preference for the shorter `GPUTextureViewDimension textureDimension`.

Alternatively, we could maybe make it default to `2D`?

Reply:
Thanks for clarification! Removed `?` now.
I don't want to default to `2D` for now, since our bind group descriptor doesn't default either (although, texture descriptor does)